mpl_interactions/xarray_helpers.py

```diff
@@ -0,0 +1,159 @@
+import numpy as np
+
+
+def choose_fmt_str(dtype=None):
+    """
+    Choose the appropriate string formatting for different dtypes.
+
+    Paramters
+    ---------
+
+    dtype: np.dtye
+        Dtype of array containing values to be formatted.
+
+
+    Returns
+    -------
+
+    fmt: str
+        Format string
+
+    """
+    if np.issubdtype(dtype, "float"):
+        fmt = r"{:0.2f}"
+    elif np.issubdtype(dtype, "int"):
+        fmt = r"{:d}"
+    else:
+        fmt = r"{:}"
+    return fmt
+
+
+def choose_datetime_nonsense(arr, timeunit="m"):
+    """
+    Try to do something reasonable to datetimes and timedeltas.
+
+    Parameters
+    ----------
+
+    arr: np.array
+        Array with values to be formatted.
+
+    Returns
+    -------
+
+    out: np.array
+        Array modified to format decently in a slider.
+
+    """
+    if np.issubdtype(arr.dtype, "datetime64"):
+        # print('datetime')
+        out = arr.astype(f"datetime64[{timeunit}]")
+    elif np.issubdtype(arr.dtype, "timedelta64"):
+        out = arr.astype(f"timedelta64[{timeunit}]").astype(int)
+    else:
+        out = arr
+    return out
+
+
+def get_hs_axes(xarr, is_color_image=False, timeunit="m"):
+    """
+    Read the dims and coordinates from an xarray and construct the
+    axes argument for hyperslicer. Called internally by hyperslicer.
+
+    Parameters
+    ----------
+
+    xarr: xarray.DataArray
+        DataArray being viewed with hyperslicer
+
+    is_color_image: bool default False
```

Comment:
Let's follow how mapltolib lists defaults. So this (and hte others like it) would become:
`is_color_image : bool, default: False`

also for all of the new docstrings you need a space between the variable name and the colon.

Reply:
otherwise they don't render well on readthedocs + is inconsistent